src/main/java/teammates/ui/automated/DatastoreBackupAction.java

```diff
@@ -0,0 +1,82 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import com.google.appengine.api.appidentity.AppIdentityServiceFactory;
+
+import teammates.common.util.Config;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.Logger;
+
+/**
+ * Cron job: performs Datastore backup.
+ *
+ * @see <a href="https://cloud.google.com/datastore/docs/export-import-entities">https://cloud.google.com/datastore/docs/export-import-entities</a>
+ * @see <a href="https://cloud.google.com/datastore/docs/reference/admin/rest/v1/projects/export">https://cloud.google.com/datastore/docs/reference/admin/rest/v1/projects/export</a>
+ */
+public class DatastoreBackupAction extends AutomatedAction {
+
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    protected String getActionDescription() {
+        return null;
+    }
+
+    @Override
+    protected String getActionMessage() {
+        return null;
+    }
+
+    @Override
+    public void execute() {
+        if (Config.isDevServer()) {
+            log.info("Skipping backup in dev server.");
+            return;
+        }
+        if (!Config.ENABLE_DATASTORE_BACKUP) {
+            log.info("Skipping backup by system admin's choice.");
+            return;
+        }
+        List<String> scopes = new ArrayList<>();
+        scopes.add("https://www.googleapis.com/auth/datastore");
+        String accessToken = AppIdentityServiceFactory.getAppIdentityService().getAccessToken(scopes).getAccessToken();
+        String appId = Config.getAppId();
+
+        HttpPost post = new HttpPost("https://datastore.googleapis.com/v1/projects/" + appId + ":export");
+        post.setHeader("Content-Type", "application/json");
+        post.setHeader("Authorization", "Bearer " + accessToken);
+
+        Map<String, Object> body = new HashMap<>();
+        // Documentation is wrong; the param name is output_url_prefix instead of outputUrlPrefix
+        body.put("output_url_prefix", "gs://" + Config.GCS_BUCKETNAME + "/datastore-backups/" + Instant.now().toString());
+
+        StringEntity entity = new StringEntity(JsonUtils.toJson(body), Charset.forName("UTF-8"));
+        post.setEntity(entity);
+
+        try (CloseableHttpClient client = HttpClients.createDefault();
+                CloseableHttpResponse resp = client.execute(post);
+                BufferedReader br = new BufferedReader(new InputStreamReader(resp.getEntity().getContent()))) {
+            String output = br.lines().collect(Collectors.joining(System.lineSeparator()));
+            log.info("Backup request successful:" + System.lineSeparator() + output);
```

Comment:
We should check the response status code?

![image](https://user-images.githubusercontent.com/13977235/49325948-d652b480-f585-11e8-8063-b5065e81f5b0.png)